content/docs/thinking-in-react.md

```diff
@@ -31,116 +31,116 @@ Our JSON API returns some data that looks like this:
-  2. **`SearchBar` (blue):** receives all *user input*
-  3. **`ProductTable` (green):** displays and filters the *data collection* based on *user input*
-  4. **`ProductCategoryRow` (turquoise):** displays a heading for each *category*
-  5. **`ProductRow` (red):** displays a row for each *product*
+  1. **`FilterableProductTable` (नारंगी):** उदाहरण का सम्पूर्णता शामिल
+  2. **`SearchBar` (नीला):** सभी *उपभोक्ता इनपुट* को लेता है
+  3. **`ProductTable` (हरा):** *यूजर इनपुट* के आधार पर *डाटा कलेक्शन* को प्रदर्शित और फ़िल्टर करता है
+  4. **`ProductCategoryRow` (तुरकोईस):**  हर *श्रेणी* के शीर्षक का प्रदर्शन
+  5. **`ProductRow` (लाल):** हर *प्रोडक्ट* की पंक्ति का प्रदर्शन
 
-If you look at `ProductTable`, you'll see that the table header (containing the "Name" and "Price" labels) isn't its own component. This is a matter of preference, and there's an argument to be made either way. For this example, we left it as part of `ProductTable` because it is part of rendering the *data collection* which is `ProductTable`'s responsibility. However, if this header grows to be complex (e.g., if we were to add affordances for sorting), it would certainly make sense to make this its own `ProductTableHeader` component.
+अगर आप `ProductTable` पर नज़र डालेंगे, तो आप देखेंगे कि टेबल हैडर ("Name" और "Price" नाम युक्त) अपना खुद का कौम्पोनॅन्ट नहीं है। यह पसंद पर निर्भर करता है और कैसे रखना है, यह चर्चा का विषय है। उदाहरण के तौर पर, हमने इसे `ProductTable` का ही भाग रखा है क्युकी यह *डाटा समूह* कि रेंडरिंग का भाग है, जो कि `ProductTable` की जिम्मेदारी है। मगर, यदि यह हैडर कठिन बनता जाता है (उदहारण के तौर पर, उसमे श्रेणीकरण की संभावना जोड़ना), तो उसका अपना खुद का कौम्पोनॅन्ट, `ProductTableHeader`, बनाना उचित होगा।
 
-Now that we've identified the components in our mock, let's arrange them into a hierarchy. Components that appear within another component in the mock should appear as a child in the hierarchy:
+अब जब हमे अपने मॉक कौम्पोनॅन्ट की पहचान हो गई है, उन्हें अब हम पदक्रम में जमाएंगे। मॉक में, जो कौम्पोनॅन्ट दूसरे कौम्पोनॅन्ट के भीतर दिखाई दे रहे है, उन्हें पदक्रम में उनके चाइल्ड की तरह दिखाना होगा:
 
   * `FilterableProductTable`
     * `SearchBar`
     * `ProductTable`
       * `ProductCategoryRow`
       * `ProductRow`
 
-## Step 2: Build A Static Version in React {#step-2-build-a-static-version-in-react}
+## स्टेप 2: React में स्थिर वर्शन बनाए {#step-2-build-a-static-version-in-react}
 
-<p data-height="600" data-theme-id="0" data-slug-hash="BwWzwm" data-default-tab="js" data-user="lacker" data-embed-version="2" class="codepen">See the Pen <a href="https://codepen.io/gaearon/pen/BwWzwm">Thinking In React: Step 2</a> on <a href="https://codepen.io">CodePen</a>.</p>
+<p data-height="600" data-theme-id="0" data-slug-hash="BwWzwm" data-default-tab="js" data-user="lacker" data-embed-version="2" class="codepen">पेन को देखे <a href="https://codepen.io/gaearon/pen/BwWzwm">React में सोचना: स्टेप २</a>, <a href="https://codepen.io">कोडपेन </a>पर।</p>
```

Comment:
`/a>, <a ` => `/a> <a `

Reply:
I did this deliberately, to make it look like -

> React में सोचना: स्टेप 2, कोडपेन पर।

wherein  `React में सोचना: स्टेप 2` will be a link.
What do you suggest?

Reply:
I was suggesting just to remove the comma `,` between both the `a` tags.
Actually, now when I check it again. It looks fine. Lets leave this as it is